annotation-file-utilities/src/annotator/find/ASTPathCriterion.java

```diff
@@ -910,6 +910,7 @@ private boolean checkTypePath(int i, Tree typeTree) {
         ++i;
       }
     } catch (RuntimeException ex) {
+      throw ex;
```

Comment:
It seems intentional that the exception isn't throw.    This method is called in `checkNull` which is called by `isSatisfiedBy`.   The comment above the call in `isSatifiedBy` says:
 ```
      // Based on the child selector and (optional) argument in "astNode",
      // "next" will get set to the next source node below "actualNode".
      // Then "next" will be compared with the node following "astNode"
      // in "actualPath". If it's not a match, this is not the correct
      // location. If it is a match, keep going.
```
So I think if its not a match, then an exception could be thrown. In that case, this method catch the exception and return false. (Or replace the catch block with a finally block.)  Also you can remove the warning suppression on this method.   

But I don't really know this code, so if you think it's safe to throw the exception, that's fine with me.

Reply:
Thanks for catching that!